[RHPAM-1179] Align EJB Timer configurations

ruromero opened the pull request:
Signed-off-by: Ruben Romero Montes <rromerom@redhat.com>

Fix https://issues.jboss.org/browse/RHPAM-1179

Behavioural changes:
* MySQL 
  * Allows receiving a `XA_CONNECTION_PROPERTY_URL` property
  * `PinGlobalTxToPhysicalConnection` is set using a property instead of the URL
* ServerName and PortNumber are assigned with the following preference:
  1. `DB_XA_CONNECTION_PROPERTY_ServerName`
  1. `DB_SERVICE_HOST`
  1. `DB_MYSQL_SERVICE_HOST`
* `NON_XA` is not forced: URL cannot be considered a valid XA property. Some drivers do not support it.
* URL is not taken from the NONXA configuration following the previous reasoning.
* Both options are configured in the same way. `DB_SERVICE_PREFIX_MAPPING` and `DATASOURCES`
* Removed configuration already set by the EAP datasource scripts.
* Removed custom configurations for `db2` and `postgresql`

ruromero commented:
@spolti do you think it's worth it adding this clean up? There were some discrepancies I have tried to align. Thanks in advance

spolti commented:
I am ok with it if all the tests are passing.
Just wondering, why did you remove the db2 tests?

ruromero commented:
@spolti I removed only the one checking for the default value
BTW I changed the topic to fix RHPAM-1179 instead as it required some changes. Can you please review?

spolti commented:
changes looks good, but before merge please make sure you can start a timer process using all supported databases.

spolti commented:
@errantepiphany I didn't have time to test it yet. Before merge it will be great to test, if possible all databases, or at least the most important like oracle and postgresql (due the problems we had in the past)

spolti commented:
The changes broke all datasources tests.
Also, the jndi name for xa-datasources is not correctly set: "<xa-datasource jndi-name="_EJBTimer""

spolti commented:
replaced by https://github.com/jboss-container-images/jboss-kie-modules/pull/107
